### PR TITLE
splunk add-on documentation

### PR DESCRIPTION
**Pull request by dantavori**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/29616

## Description
splunk add-on documentation

### CODE REVIEW

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+id: splunk-add-on
+title: Demisto Add-on for Splunk
+---
+Supporting Add-on for Cortex XSOAR. This application allows to push incidents into Cortex XSOAR from Splunk according to configurable trigger parameters.
+
+
+### Prerequisites
+* Splunk version 7.2 >=
+
+* In order to use a Splunk Cloud instance, you need to contact Splunk support and ask them to open API access and use the non SAML authentication with it, as stated in this message from Splunk:
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:11]: @dantavori Are you sure about this?

**Comment by glicht** [comment]
Looks like you have invalid image urls:

image

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  > Use the following URL for Splunk Cloud deployments. If necessary, submit a support case to open port 8089 on your deployment. Please include the IP Addresses/CIDR Ranges you would like to have access from.<br/>
+    https://\<deployment-name\>.splunkcloud.com:8089
+  
+  For additional information see: https://docs.splunk.com/Documentation/SplunkCloud/latest/RESTTUT/RESTandCloud
+
+
+### Prepare a local Splunk Environment
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:18]:
```suggestion
### Test on a local Splunk Environment
```

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+
+### Installation of the Add-on
+* Download Demisto Add-on for Splunk from [Splunkbase](https://splunkbase.splunk.com/app/3448).
+* After initializing the container, open your local Splunk environment.
+* Go to "Manage Apps" → Install app from file → upload the latest version of Demisto Add-on for Splunk.
+  *Note:* if a version of the app already exists, mark the "Upgrade app" checkbox.
+  
+  ![install](https://user-images.githubusercontent.com/38749041/103541256-db406b80-4ea3-11eb-9280-279f50e447f6.gif)
+* Restart Splunk and login again.
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:33]: @dantavori  Is this really required?

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+       - By default, "Validate SSL" is enabled.
+       
+       ![image](https://user-images.githubusercontent.com/38749041/103541559-4722d400-4ea4-11eb-8b01-754d9edd570c.png)
+* You must restart Splunk in order to apply changes in the configuration settings.
+
+       
+### Connectivity Test - Create a Custom Alert Action from Saved Searches
+* Upload data to Splunk (any small PDF, CSV or YML file).
+
+  ![alert](https://user-images.githubusercontent.com/38749041/103539271-6d467500-4ea0-11eb-89f9-2a551893800f.gif)
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:57]: gif doesn't show up well (on my env at least). Please check

image

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      3. XSOAR Server (if "Send Alert to all the servers" is unchecked)
+      4. Type – incident type in XSOAR
+      5. Custom Fields - A comma separated 'key:value' custom fields pairs
+      6. Labels – a comma separated values to be put in the labels field
+      7. Severity – the alert severity
+      8. Details – "details" field of the incident
+
+* Go to the XSOAR server and wait for incidents (one for each event in Splunk).
+
+  ![image](https://user-images.githubusercontent.com/38749041/103539782-52c0cb80-4ea1-11eb-94a3-e284a97b33f3.png)
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:77]: Image is practically impossible to read

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+### Troubleshooting
+We recommend checking the following logs as an initial step in troubleshooting:
+- `var/log/splunk/splunkd.log` – for Splunk issues
+- `var/log/splunk/create_xsoar_incident_modalert.log` – for the add-on issues
+
+
+### Common Issues
+
+#### Splunk Events are not Created in Demisto
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:112]:
```suggestion
#### Splunk Events are not Created in Cortex XSOAR
```

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+In cases where after associating Create Demisto Incident with saved searches or correlation searches using automated invocation or using ad-hoc invocation from Splunk-ES incident review dashboard, Incidents are not getting created into the Cortex XSOAR, the following criteria should be checked.
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:113]:
```suggestion
In cases where after associating Create XSOAR Incident with saved searches or correlation searches using automated invocation or using ad-hoc invocation from Splunk-ES incident review dashboard, Incidents are not getting created into the Cortex XSOAR, the following criteria should be checked.
```

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+1. Check the configuration page. You may need to restart for the modifications to take effect.
+
+2. Check the Saved Search trigger conditions, Splunk might be suppressing the events and not creating an incident in Cortex XSOAR.
+
+3. Test the network connectivity in between applications to ensure that there are no connectivity issues. You may try one of the following:
+   ```
+   curl -k https://<DEMISTO_SERVER>:<PORT>/proxyMode
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:121]: Not clear what is `/proxyMode` needed for. Also add -v to the curl command.

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+   telnet <DEMISTO_SERVER> <PORT>
+   wget --no-check-certificate -O - https://<DEMISTO_SERVER>:<PORT>/proxyMode
+   ```
+
+4. Test the API key generated in XSOAR.
+   ```
+   wget --no-check-certificate --header="Authorization: <API_KEY>" -O - https://<DEMISTO_SERVER>:<PORT>/user
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:128]: Better to use curl as an example.

File: content-repo/extra-docs/articles/splunk-add-on.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+#### Incident Created with Incorrect Fields
+If incidents being created in XSOAR but with wrong fields, try to refer to the [Connectivity Test](#connectivity-test---create-a-custom-alert-action-from-saved-searches) section for an understanding of the fields which are associated with create incident actions.
+
+#### Too Many Incidents Created in XSOAR
+Splunk has its own trigger conditions using the number of incidents which get pushed to XSOAR controlled.
+
+Refer to [Splunk Documentation](https://docs.splunk.com/Documentation/Splunk/latest/Alert/AlertTriggerConditions) to understand Splunk Trigger Conditions to limit the incidents. 
+In addition, it is possible to [throttle alerts](https://docs.splunk.com/Documentation/SplunkCloud/latest/Alert/ThrottleAlerts) in Splunk once the saved search returned with positive results, in order to suppress more incident creation.
+
+#### Playbooks not getting invoked on Demisto Enterprise Application
```
Comment (glicht) [inline content-repo/extra-docs/articles/splunk-add-on.md:140]:
```suggestion
#### Playbooks not getting invoked on Cortex XSOAR
```

File: content-repo/extra-docs/articles/splunk-add-on.md (hunk 3 above)
Comment (dantavori) [reply content-repo/extra-docs/articles/splunk-add-on.md:27]: yes, you are redirected to this page after submitting the add-on file: 
image

**Comment by glicht** [comment]
@kirbles19 please review

**Comment by kirbles19** [comment]
@dantavori Done

**Review by glicht — APPROVED** [review]

**Pull request merged**